### PR TITLE
Updated index.html with lots of sweet goodness

### PR DESCRIPTION
**Pull request by henshaw**
1. Removed the title attributes from links because they are useless. Useless for SEO or any other purpose. And they aren't any good for accessibility either: https://www.a11yproject.com/posts/title-attributes/
2. Modified the dfn attribute, which should really be wrapped around the word. The way the sentence is structured, parsers and decent algos will be able to comprehend word and its definition as it's written.
3. Replaced all of the ’ with ' because gross.
4. Removed aside element, because it's really part of the article. It's not a true aside and is therefore not being used semantically as intended. If you wanted to use an aside element, it would be used to reference something else that's vaguely related to the article. But this is completely relevant to the article.
5. Almost every paragraph didn't have a closing p element. What kind of sick bastard leaves their paragraph wide open for anyone to tickle?! Fixed!
6. Added a bunch of abbr elements to the acronyms and abbreviations. Unlike title attributes on links, these are actually useful. They're helpful to readers who aren't familiar with an acronym or abbreviation and they disambiguate the meaning for search engines (that's a good thing). 
7. Added canonical link. Kind of helpful considering this is a public repos and anyone could post it anywhere. You want that hint to be there for search bots to consider this domain the end all, be all domain.
8. Added a better meta for UTF-8 content type
9. All this talk on the page about HTTPS and SEO but no meta description. For shame!
10. Added Open Graph data because you want to be social, don't you? DON'T YOU!? I made an Open Graph image for it. Feel free to change it to whatever you want, but it needs to exist and should be no smaller than 1200x630 px. It's on IPFS, just for the heck of it. Let's live in the future, but feel free to add it to the replace if you want.
11. Added Schema structured data in JSON format. That's the format Google prefers and you should disambiguate ALL THE THINGS and talk them bots in their own language.
12. Added periods to the headings that didn't have them. Consistency would make mother proud.
13. Added system fonts so the main system font for the OS would be used. If that's not elegant as fuck, I don't know what is. It's instantaneous and nothing is loaded because it already IS loaded. It's also consistent with the paragraph about using fonts on the page.

### CODE REVIEW

**Comment by jtagcat** [comment]
1. Broken in to https://github.com/KeenRivals/bestmotherfucking.website/issues/66

**Comment by jtagcat** [comment]
Please do break the PR up to multiple parts for the sanity of others and yourself. It honestly looks bad.

File: index.html [hunk 1]
```diff
@@ -1,63 +1,82 @@
...
-		padding:0 .62em;
-		font:1.2em/1.62 sans-serif
-	}
-	h1,h2,h3 {
-		line-height:1.2
+	@media (prefers-color-scheme: dark) {
+		body { color:#fff;background:#000; }
+		a:link { color:#cdf; }
+		a:hover, a:visited:hover { color:#def; }
+		a:visited { color:#dcf; }
```
Comment (jtagcat) [inline index.html:23]: There was saving of bytes going on, and you are undoing the changes.

File: index.html [hunk 2]
```diff
@@ -1,63 +1,82 @@
...
 	}
+	body{ margin:1em auto;max-width:40em;padding:0 .62em;font:1.2em/1.62 sans-serif;font-display:optional;font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue", sans-serif; }
```
Comment (jtagcat) [inline index.html:25]: Please do use newlines for readability.

I'm also pretty sure leaving the fonts change out is preferred, see other issues and discussions.

File: index.html [hunk 3]
```diff
@@ -1,63 +1,82 @@
...
 <link rel="icon" type="image/png" href="data:image/png;base64,">
+<link rel="canonical" href="https://bestmotherfucking.website">
 <title>Best Motherfucking Website</title>
+<meta name="description" content="This website is fucking perfect. Seriously, what the fuck else do you want?">
+<meta property="og:type" content="website" />
+<meta property="og:title" content="Best Motherfucking Website" />
+<meta property="og:description" content="This website is fucking perfect. Seriously, what the fuck else do you want?" />
+<meta property="og:url" content="https://bestmotherfucking.website" />
+<meta property="og:site_name" content="Best Motherfucking Website" />
+<meta property="og:image" content="https://ipfs.io/ipfs/QmeVTyCq1JoN6Vhv5RQxWU28Lg6cQd3E273RwVwQpbX7Be?filename=og.png" />
```
Comment (jtagcat) [inline index.html:14]: ipfs: no thanks

Comment (jtagcat) [reply index.html:14]: also wtf is that image even, stop

**Comment by jtagcat** [comment]
5. You are undoing previous changes: https://github.com/KeenRivals/bestmotherfucking.website/commit/50990d40870a7ea6e72966541444e4f2abfa6972

File: index.html (hunk 3 above)
Comment (jtagcat) [reply index.html:14]: The image adds nothing, it's an external dependency, it's bytes.

It actually makes the preview worse, take up more space!

image

image

**Comment by jtagcat** [comment]
3. No, smart commas are a thing.

**Comment by jtagcat** [comment]
4. uhh I don't really agree, third opinion would be needed; I'd say it's not a bad advertisement of 'btw other html tags exist'

**Comment by jtagcat** [comment]
> 6. Added a bunch of abbr elements to the acronyms and abbreviations. Unlike title attributes on links, these are actually useful. They're helpful to readers who aren't familiar with an acronym or abbreviation and they disambiguate the meaning for search engines (that's a good thing).

Make it a separate Issue/PR. Though:
 - abbr / definitions are used for the first occurrence 
 - byte is a commonly used SI (standard) unit like meters, (kilo)gram, watts, the reader very likely knows what it means, especially the target audience here.
 - other abbrs are also quite common, people can google it; almost all of it looks like spam from you

File: index.html (hunk 3 above)
Comment (jtagcat) [reply index.html:14]: edit:

> Let's live in the future

no thanks, here, old boring is better

**Comment by jtagcat** [comment]
> 11. Added Schema structured data in JSON format. That's the format Google prefers and you should disambiguate ALL THE THINGS and talk them bots in their own language.

No. Fuck Google. 

I can't think of any, but there might be very few edge cases. It might have an a11y element for screen readers or for stuff extracting content (say Wallabag (actually a lib it depends on)). In any case, it's likely neither of them don't use it.

HTML structure and tags (ex `<article>`) do it just fine. All three (a11y, content readers, search engines) work just fine.

**Comment by jtagcat** [comment]
> Added periods to the headings that didn't have them. Consistency would make mother proud.

Titles don't have periods.

**Comment by jtagcat** [comment]
Good morning to you as well. https://xkcd.com/386/

**Comment by henshaw** [comment]
😂

**Comment by KeenRivals** [comment]
1. Interesting point on title attributes. I will look into / consider that for a future update. If you're willing to PR the fix that might make things easier for me.
2. I am realizing I might be doing the dfn thing wrong. I might just remove the dfn thing.
3. I like smart quotes.
4. I'm using aside as kind of a sub-title.
5. Closing tags are optional in certain elements in html5.
6. I don't really like littering the abbrs everywhere. I expect my audience to know what they are or be capable of using a search to find out.
7. Canonical link might be a good idea. I'll have to look into that.
8. The server is already sending that Content-Type header anyways.
9. This site is so simple that search engines do great on their own figuring it out.
10. What does open graph do for this site that search engines are not doing already? I don't really want a screenshot.
11. One of the points of this site is that if you keep your site reasonably simple, bots do all this work for you.
12. I don't like periods in headings (generally).
13. Specifying the generic sans-serif font lets systems use their font of choice, which users can override.

Also, appreciate you @jtagcat

**Pull request closed without merging**